### PR TITLE
no-op refactor: extract outLink generation into a function

### PR DESCRIPTION
**Pull request by mupdt**
# Motivation

We are working on the #7890 feature. As we're working on it, we are identifying non-controversial/no-op cleanups an extracting them into small pull requests. This is the first such cleanup pull request.

The idea is to make the final PR for the #7890 feature smaller but also to contribute even if the #7890 feature doesn't get merged. I hope that makes sense. Let me know if this approach is not okay.

Maintainers: tick if completed or explain if not relevant

 - [ ] agreed on idea
 - [ ] agreed on implementation strategy
 - [ ] tests, as appropriate
   - functional tests - `tests/**.sh`
   - unit tests - `src/*/tests`
   - integration tests - `tests/nixos/*`
 - [ ] documentation in the manual
 - [ ] code and comments are self-explanatory
 - [ ] commit message explains why the change was made
 - [ ] new feature or incompatible change: updated release notes

cc: @rickynils

### CODE REVIEW

File: src/nix/build.cc [hunk 1]
```diff
@@ -41,6 +41,32 @@ nlohmann::json builtPathsWithResultToJSON(const std::vector<BuiltPathWithResult>
     return res;
 }
 
+void createOutLinks(const Path& outLink, const std::vector<BuiltPathWithResult>& buildables, ref<Store>& store) {
+    auto store2 = store.dynamic_pointer_cast<LocalFSStore>();
```
Comment (Ericson2314) [inline src/nix/build.cc:45]: I think it is better to do the downcast in the caller.
```suggestion
void createOutLinks(const Path& outLink, const std::vector<BuiltPathWithResult>& buildables, LocalFSStore & store)
{
```
Also we usually do the opening brace on the next line.

**Comment by Ericson2314** [comment]
There is code in 

- `src/libcmd/repl.cc`
- `src/nix-store/nix-store.cc`
- `src/nix-build/nix-build.cc`

that really ought to be deduplicated with this too. If you can do get all those, pulling out function(s) will be all the more valuable!

File: src/nix/build.cc (hunk 1 above)
Comment (mupdt) [reply src/nix/build.cc]: The opening brace is now on a new line and the downcast is at the call-site.

**Comment by mupdt** [comment]
@Ericson2314: I gave it a try with the deduplications you mentioned, but the code is sufficiently different that I wasn't confident I can do a no-op extraction. I hope that's okay.

File: src/nix/build.cc [hunk 2]
```diff
@@ -41,6 +41,28 @@ nlohmann::json builtPathsWithResultToJSON(const std::vector<BuiltPathWithResult>
     return res;
 }
 
+void createOutLinks(const Path& outLink, const std::vector<BuiltPathWithResult>& buildables, LocalFSStore& store2)
```
Comment (Ericson2314) [inline src/nix/build.cc:44]:
```suggestion
static void createOutLinks(const Path& outLink, const std::vector<BuiltPathWithResult>& buildables, LocalFSStore& store2)
```

**Comment by Ericson2314** [comment]
Fair enough, thanks for trying.

**Review by roberth — APPROVED** [review]

**Pull request merged**